independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanInfo.java

```diff
@@ -518,8 +518,11 @@ protected String getType() {
             for (Entry<MethodKey, Set<AnnotationInstance>> entry : candidates.entrySet()) {
                 List<InterceptorInfo> interceptors = beanDeployment.getInterceptorResolver()
                         .resolve(InterceptionType.AROUND_INVOKE, entry.getValue());
-                if (!interceptors.isEmpty()) {
-                    interceptedMethods.put(entry.getKey().method, new InterceptionInfo(interceptors, entry.getValue()));
+                MethodInfo method = entry.getKey().method;
+                // @PreDestroy and @PostConstruct methods declared on the bean are NOT candidates for around invoke interception
+                if (!interceptors.isEmpty() && !method.hasAnnotation(DotNames.PRE_DESTROY)
```

Comment:
Woudln't it make more sense to ignore such methods in the `io.quarkus.arc.processor.Methods.SubclassSkipPredicate.test(MethodInfo)` instead?

Reply:
Yea, that's another way. I can rework it if you want to.
Either way, I think we might run into a corner case problem where some users might attempt to call these methods manually (which interceptors spec doesn't forbid I think?) and at that point they still won't be intercepted, But it is IMO still better than what we do now.

Reply:
Changed accordingly to your suggestion.